common/src/config.rs

```diff
@@ -22,6 +22,26 @@ impl HostPort {
     pub fn to_string(&self) -> String {
         format!("{}:{}", self.host, self.port)
     }
+
+    pub fn to_socket_addr(&self) -> std::net::SocketAddr {
```

Comment:
I think this is doing too much to be on the config. The configs should just be data structs, name resolution should happen elsewhere.
Also, let's not unwrap and panic, and instead return an error

Reply:
Both sound good. Here's the current situation with name resolution:
- The standard library defines a trait, `ToSocketAddrs` (https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html). It's defined on some types like `(&str, u16)`.
- I can perhaps define it for the HostPort implementation and basically just pass it to the standard library.
- We could also have a separate helper function that takes a hostport and returns a SocketAddr.

Which one would you prefer? Or a different one if not listed above.

Reply:
implementing ToSocketAddrs trait for HostPort sounds good to me, assuming there's a way to return an error from there right?

Reply:
There is. Implemented that solution in a new commit on top.